src/menu/MenuSongSelection.as

```diff
@@ -868,6 +873,36 @@ package menu
                 }
             }
         }
+        
+        /**
+         * Song Item Context Menu: Plays the chart preview of the selected song.
+         */
+        private function e_playChartPreviewContextSelect(e:ContextMenuEvent):void
+        {
+            if (!_gvars.options)
+            {
+                _gvars.options = new GameOptions();
+                _gvars.options.fill();
+            }
+            _gvars.options.replay = new SongPreview((e.contextMenuOwner as SongItem).level);
+            _gvars.options.loadPreview = true;
+
+            if (!_gvars.options.replay.isLoaded)
+            {
+                (_gvars.options.replay as SongPreview).setupSongPreview();
+            }
+
+            if (_gvars.options.replay.isLoaded)
+            {
+                // Setup Vars
```

Comment:
There's a possible soft-lock possible here if you don't force the judge and global offsets to be 0. Normally these are zero for the on site preview as user data is skipped. Just toss
```
// Force no offsets on judge.
_gvars.options.offsetGlobal = 0;
_gvars.options.offsetJudge = 0;
```
before the `// Setup Vars`